ebean-core/src/main/java/io/ebeaninternal/server/deploy/BeanDescriptor.java

```diff
@@ -3038,6 +3038,10 @@ public boolean isReference(EntityBeanIntercept ebi) {
     return ebi.isReference() || referenceIdPropertyOnly(ebi);
   }
 
+  boolean hasIdPropertyOnly(EntityBeanIntercept ebi) {
```

Comment:
This method should not be required and instead use the existing `referenceIdPropertyOnly()`.  That takes into account generated properties - idOnlyReference true when there is other non-generated scalar properties on the bean.